File: java/etl/src/main/java/Etl.java
import java.util.AbstractMap;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class Etl {
    public Map<String, Integer> transform(Map<Integer, List<String>> old) {

        HashMap<String, Integer> mapByLetter = new HashMap<String, Integer>();
        old.forEach( (score, letters) ->
            letters.stream().forEach(letter -> mapByLetter.put(letter.toLowerCase(), score) )
        );
        return mapByLetter;
    }
}

File: elixir/etl/etl.exs
defmodule ETL do
  @doc """
  Transform an index into an inverted index.

  ## Examples

  iex> ETL.transform(%{"a" => ["ABILITY", "AARDVARK"], "b" => ["BALLAST", "BEAUTY"]})
  %{"ability" => "a", "aardvark" => "a", "ballast" => "b", "beauty" =>"b"}
  """
  @spec transform(map) :: map
  def transform(input) do
    input
    |> Enum.reduce(%{}, fn( {score, letters}, acc) ->
      letters
      |> Enum.reduce(%{}, fn(letter, acc2) ->
        acc2 |> Map.put(letter |> String.downcase, score)
      end)
      |> Map.merge(acc)
    end)
  end
end